keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/BetaVertex.java

```diff
@@ -108,6 +110,30 @@ public double logProb(DoubleTensor value) {
         return logPdfs.sum();
     }
 
+    @Override
+    public LogProbGraph logProbGraph() {
```

Comment:
I also think that it should be the distribution that implements the `LogProbGraphSupplier` interface, rather than the vertex.

Reply:
This is a good comment. If we are keeping distribution classes in the long run, it'll be worth making this separation now.

Reply:
The distribution class could just take placeholder vertices as parameters, to ensure it doesn't mutate the original network

Reply:
Talked about it offline. Will consider this in the future.